app/controllers/landing_page_controller.rb

```diff
@@ -7,9 +7,15 @@ def search
     @form = SearchForm.new(params[:search_form])
 
     if @form.valid?
-      @result = FirmRepository.new.search(@form.to_query)
+      @result = FirmRepository.new.search(@form.to_query, page: page)
     else
       render :show
     end
   end
+
+  private
+
+  def page
+    params[:page].try(:to_i) || 1
```

Comment:
For me is fine this line, but following the Hash#fetch idea, instead using ||, could be like params.fetch(:page, 1).to_i 
But this two could sound a little weird. :)